pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -972,6 +972,22 @@ private void sendFlowPermitsToBroker(ClientCnx cnx, int numMessages) {
             if (log.isDebugEnabled()) {
                 log.debug("[{}] [{}] Adding {} additional permits", topic, subscription, numMessages);
             }
+
+            cnx.ctx().writeAndFlush(Commands.newFlow(consumerId, numMessages))
+                    .addListener(writeFuture -> {
+                        String msg;
+                        if (!writeFuture.isSuccess()) {
+                            msg = String.format("Consumer {} failed to send {} permits to broker: {}"
+                                    , consumerId, numMessages, writeFuture.cause().getMessage());
+                            increaseAvailablePermits(cnx, numMessages, false);
+                        } else {
+                            msg = String.format("Consumer {} sent {} permits to broker", consumerId, numMessages);
+                        }
+                        if (log.isDebugEnabled()) {
+                            log.debug(msg);
+                        }
+                    });
+
             if (log.isDebugEnabled()) {
```

Comment:
Do we need to remove line 991~1000?

Reply:
+1, we can't send duplicated Flow command.

Reply:
Fixed.